build.psm1

```diff
@@ -679,22 +679,77 @@ function Start-PSxUnit {
 }
 
 
+function Install-Dotnet {
+    [CmdletBinding()]
+    param(
+        [string]$Channel = "rel-1.0.0",
+        [string]$Version = "latest",
+        [switch]$NoSudo
+    )
+
+    $obtainUrl = "https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain"
+
+    # Install for Linux and OS X
+    if ($IsLinux -or $IsOSX) {
+        # Uninstall all previous dotnet packages
+        $uninstallScript = if ($IsUbuntu) {
+            "dotnet-uninstall-debian-packages.sh"
+        } elseif ($IsOSX) {
+            "dotnet-uninstall-pkgs.sh"
+        }
+
+        if ($uninstallScript) {
+            Start-NativeExecution {
+                curl -sO $obtainUrl/uninstall/$uninstallScript
+                Invoke-Expression "$sudo bash ./$uninstallScript"
```

Comment:
Oh, you're relying on dynamic scoping, the explanation is later in the file, and you don't check -NoSudo in this function.

Would it be cleaner to introduce a helper function for `sudo` so you can avoid relying on dynamic scoping and use `&` instead of `Invoke-Expression` by testing `-NoSudo`?